app/app.go

```diff
@@ -642,6 +643,35 @@ func wireValidatorMock(conf Config, pubshares []eth2p0.BLSPubKey, sched core.Sch
 		return nil
 	})
 
+	go func() {
+		for registration := range conf.TestConfig.BuilderRegistration {
+			ctx := log.WithTopic(context.Background(), "vmock")
+
+			addr := "http://" + conf.ValidatorAPIAddr
+			eth2Svc, err := eth2http.New(ctx,
+				eth2http.WithLogLevel(1),
+				eth2http.WithAddress(addr),
+				eth2http.WithTimeout(time.Second*10), // Allow sufficient time to block while fetching duties.
+			)
+			if err != nil {
+				log.Error(ctx, "Cannot connect to validatorapi", err)
+				return
+			}
+			eth2Cl, ok := eth2Svc.(validatormock.Eth2Provider)
+			if !ok {
+				log.Error(ctx, "Invalid eth2 service", nil)
+			}
+
+			// what pubkey should this be - should we be handing in registrations & pubkeys
+			err = validatormock.Register(ctx, eth2Cl, signer, registration, pubshares[0])
```

Comment:
This is receiving registrations from the chan - i've just used pubshares[0] for now 

how would you want to associate a pubkey with a registration that is received through the channel

would have used indexes but range on chan doesn't return index - not sure if theres a work around